Controllers/FileController.cs

```diff
@@ -23,27 +80,91 @@ public async Task<ActionResult> Post()
                 Directory.CreateDirectory(uploadPath);
             }
 
-            var path = Path.Combine(uploadPath, file.FileName);
+            foreach (var file in files)
+            {
+                var path = Path.Combine(uploadPath, file.FileName);
+
+                using (var stream = new FileStream(path, FileMode.Create))
+                {
+                    await file.CopyToAsync(stream);
+                }
+            }
+            string uploaderName = form["uploader"];
+            string desserts = form["dessert"];  // Assuming this is a string like "dessert1,dessert2,dessert3"
+            string weather = form["weather"];
+            string country = form["country"];
+
+            // Split the desserts string into an array of individual desserts
+            string[] dessertArray = desserts.Split(',');
 
-            using (var stream = new FileStream(path, FileMode.Create))
+            // Loop through the array and add each dessert individually
+            foreach (string dessert in dessertArray)
             {
-                await file.CopyToAsync(stream);
+                // Increment a vote counter for the type of dessert
+                if (this.dessertVotes.ContainsKey(dessert))
```

Comment:
This confused me a bit. If I want to submit a dessert vote without uploading a file, there isn't an option to do that, but there isn't client-side validation to enforce this or make it clear to the end-user. Can you help me understand the purpose of this addition?

Reply:
I was just trying to test multiple types of submission options. The idea is the whole form is submitted at once. I could make them all required. This one tests the radio input.

Reply:
This is helpful, thank you.

Reply:
These tests need to be in addition to file upload, to test if a full form submission works.